react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+export default class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      counter: 1
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({ counter: this.state.counter + 1 });
+  }
+
+  render() {
+    const count = this.state.counter;
+    const text = 'Hot Button';
+    if (this.state.counter < 4) {
+      return <button className="btn btn-indigo" onClick={this.handleClick}>
+        {text}
+      </button>;
+    } else if (count < 7) {
+      return <button className="btn btn-purple" onClick={this.handleClick}>
+        {text}
+      </button>;
+    } else if (count < 10) {
+      return <button className="btn btn-roman" onClick={this.handleClick}>
+        {text}
+      </button>;
+    } else if (count < 13) {
+      return <button className="btn btn-apricot" onClick={this.handleClick}>
+        {text}
+      </button>;
+    } else if (count < 16) {
+      return <button className="btn btn-yellow" onClick={this.handleClick}>
+        {text}
+      </button>;
+    } else if (count >= 16) {
+      return <button className="btn btn-white" onClick={this.handleClick}>
+        {text}
+      </button>;
+    }
```

Comment:
This is a lot of repetition and would be hard to maintain. It would be better to first figure out the correct class then return a single button.

```JSX
const count = this.state.counter;
const text = 'Hot Button';
let btnClass = '';
  
if (count < 4) {
  btnClass = "btn-indigo";
} else if (count < 7) {
  btnClass = "btn-purple";
} else if (count < 10) {
  btnClass = "btn-roman";
} else if (count < 13) {
  btnClass = "btn-apricot";
} else if (count < 16) {
  btnClass = "btn-yellow";
} else if (count >= 16) {
  btnClass = "btn-white";
}

return (
  <button className={`btn ${btnClass}`} onClick={this.handleClick}>
    {text}
  </button>
);
```